### DISCUSSION TITLE
LSD2: Dating ERROR: There's conflict or not enough signal in the input temporal constraints.

### DISCUSSION THREAD

**Discussion by aalonsor15** [iqtree/iqtree2#182] [Q&A] [answered]
Hello! I am using IQtree to create a tree using DNA barcoding data of moths. I want the tree constrained at the superfamily level, rooted, and dated at the common ancestor of all my sampled moths. This is the command I'm using:

iqtree2 -s my_sequences.fasta -o "LNAUT645_14_Prototheora_corvifera_CCDB_22946_G04_Hepialidae" -g constrain.newick --date age_file.txt -B 1000 -bnni -T AUTO --prefix my_timed_tree

The constrain and the rooting seems to be working, but when attempting to date, I get this error:
ERROR: There's conflict or not enough signal in the input temporal constraints.

After that error the program stops, and it never outputs a timed tree.
The age_file I am using is just a list of all of my tip labels (excluding the outgroup) separated by commas, and then -150 for 150 million years. I have attached the file here 
[age_file.txt](https://github.com/iqtree/iqtree2/files/15133484/age_file.txt)

It's formatted in this manner:
Sp1,Sp2,Sp3 -150

Here's the .log file output in case its helpful, but it doesn't show the last line where the Error appeared and the program stopped. 
[my_timed_tree.log](https://github.com/iqtree/iqtree2/files/15134924/my_timed_tree.log)

Please help! Ultimately, my goal is to produce a dated tree that I can use in R for calculating phylogenetic diversity. 

Thanks!

**Comment by bqminh** [comment]
[@tothuhien](https://github.com/tothuhien), the author of LSD2 code, might be able to answer your question.

**Comment by aalonsor15** [comment]
Thanks! 
I also tried dating an existing tree with the following command:

iqtree2 -s my_sequences.fasta --date age_file.txt -te MLtree_constrained_rooted.treefile --prefix rooted_timed_tree -nt AUTO

This does run and provides a .timetree output, but I lose the constrain I had added to the tree, as well as the root. Here's the log file: 
[rooted_timed_tree.log](https://github.com/iqtree/iqtree2/files/15152618/rooted_timed_tree.log)

At the end, I get this:
*WARNINGS
-The tips of [very long list of species, probably all of them] do not have temporal constraint, their dates will be infered.
- There's not enough input date information to have a unique solution.
The reported results show the upper solution: upper_rate and upper_dates.
For any coefficient k<=1, there's a corresponding solution where the rate is upper_rate*k,
and the date of any node i is upper_dates[i]/k

*RESULTS:
- Dating results:
 rate NA:nan, tMRCA NA:nan , objective function nan


The Dating results all say NA? So overall, I'm not sure it actually worked.

**Reply by bqminh** [reply]
I suggest that that you search this up in the iqtree Google group. I think I've seen something similar before. If you don't find the answer, please contact @tothuhien directly. I don't know how to get her involved in this discussion... The LSD2 code is cloned from her repo https://github.com/tothuhien/lsd2. So you can ask there directly.

**Comment by tothuhien** [comment, marked as answer]
Hi, not sure why I didn't get notification last time, but just get it now. It seems that you didn't specify the tip date, which is normally 0, so just try to add the option `--date-tip 0` 
Hien

**Comment by aalonsor15** [comment]
Thank you @bqminh and @tothuhien! Dating the tip was the answer! The analysis ran and the result looks promising. Here is the log: 
[multidated_tree.log](https://github.com/iqtree/iqtree2/files/15201617/multidated_tree.log)

I have another question though. It seems that the tree I get from this analysis is not fully resolved. There are a lot of polytomies. Is there a way to tell IQtree to give me the best/fully resolved tree, or tell it to forbid/resolve polytomies, or tell it to not collapse branches even though they have no clear optimal placement?

**Reply by roblanf** [reply]
@aalonsor15 once your initial question has been answered, please mark the answer (I see you did that, thanks!) and ask new questions as a new discussion item. 

This just helps to keep the discussion well organised and so a useful resource for the community. 

Thanks!

Rob

**Comment by xuxiaodong111** [comment]
I also encountered an error when running IQ-TREE version 2.3.6 with temporal constraints. 

Here’s my command:
iqtree2 -s FcC_supermatrix.fas --date DATA.txt -g Figtree.nwk -m LG -T AUTO -o "Hemiptera_species" --date-ci 100 --date-tip 0

Error Output
TREE 1
*PROCESSING:
Reading the tree ... 
Collapse 0 (over 242) internal branches having branch length <= 1e-06
 (settable via option -l)
Parameter to adjust variances was set to 0.0375 (settable via option -b)
Minimum branch length of time scaled tree (settable via option -u and -U): 0
Error: There's conflict or not enough signal in the input temporal constraints.

**Reply by tothuhien** [reply]
Hi, so there may be some conflicts between the input dates and the tree topology. If you can send me (here or through email tothuhien@gmail.com) the date file (DATA.txt) and the tree Figtree.nwk that you used in the command line above, I can have to look.
Hien
